### PR TITLE
Add labels for clusterings

### PR DESCRIPTION
**Pull request by KrisDavie**
Enable labels when searching for All Clusters.

Use community annotations to determine cluster names in the sidebar and in the labels.
This is not applied to search spaces and search results, but annotations are shown in descriptions there

### CODE REVIEW

**Comment by KrisDavie** [comment]
Example output:

image

**Review by dweemx — APPROVED** [review]

**Comment by KrisDavie** [comment]
Also closes #390

**Review by dweemx — DISMISSED** [review]
Nice work! Tested :heavy_check_mark:

File: opt/scopeserver/dataserver/utils/proto.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import json
+import hashlib
+
+from typing import Dict, Any
+
+from scopeserver.dataserver.modules.gserver import s_pb2
+
+
+def protoize_cell_type_annotation(md, secret: str):
```
Comment (MaybeJustJames) [inline opt/scopeserver/dataserver/utils/proto.py:13]: What is `md`, can you make it more descriptive and type annotate it please? Also function docstring (without types)

File: opt/scopeserver/dataserver/utils/proto.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                for cta in ctas:
+                    hash_data = json.dumps(cta["data"]) + secret
+                    data_hash = hashlib.sha256(hash_data.encode()).hexdigest()
+
+                    votes: Dict[str, Any] = {
+                        "votes_for": {"total": 0, "voters": []},
+                        "votes_against": {"total": 0, "voters": []},
+                    }
+
+                    for i in votes.keys():
```
Comment (MaybeJustJames) [inline opt/scopeserver/dataserver/utils/proto.py:28]:
```suggestion
                    for i in votes:
```

File: opt/scopeserver/dataserver/utils/proto.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                        for v in cta["votes"][i]["voters"]:
+                            hash_data = json.dumps(cta["data"]) + v["voter_id"] + secret
+                            user_hash = hashlib.sha256(hash_data.encode()).hexdigest()
+                            v["voter_hash"] = True if user_hash == v["voter_hash"] else False
```
Comment (MaybeJustJames) [inline opt/scopeserver/dataserver/utils/proto.py:32]:
```suggestion
                            v["voter_hash"] = user_hash == v["voter_hash"]
```

File: opt/scopeserver/dataserver/utils/proto.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                            votes[i]["voters"].append(s_pb2.CollabAnnoVoter(**v))
+                            votes[i]["total"] += 1
+                        votes[i] = s_pb2.CollabAnnoVotes(**votes[i])
+
+                    cta_proto = s_pb2.CellTypeAnnotation(
+                        data=s_pb2.CollabAnnoData(**cta["data"]),
+                        validate_hash=True if data_hash == cta["validate_hash"] else False,
```
Comment (MaybeJustJames) [inline opt/scopeserver/dataserver/utils/proto.py:39]:
```suggestion
                        validate_hash=data_hash == cta["validate_hash"],
```

File: opt/scopeserver/dataserver/utils/proto.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                        votes_for=votes["votes_for"],
+                        votes_against=votes["votes_against"],
+                    )
+                    proto_cell_type_annotations.append(cta_proto)
+                md[n]["clusters"][m]["cell_type_annotation"] = proto_cell_type_annotations
+    return md
+
+
+def protoize_cluster_marker_metric(metric, cluster_marker_metrics):
```
Comment (MaybeJustJames) [inline opt/scopeserver/dataserver/utils/proto.py:48]: Type annotations and function docstring?

**Review by MaybeJustJames — CHANGES_REQUESTED** [review]
Just a few minor changes then I think we're good

**Review by MaybeJustJames — APPROVED** [review]
Great!

**Pull request merged**